rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -1612,6 +1620,89 @@ private List<Map<String, Object>> getReleaseObjectMapper(List<EntityModel<Releas
 		return modifiedList;
 	}
 
+    public List<Release> filterReleases(User sw360User, String withSourceAttachment, String withoutSourceAttachment, String withCliAttachment, String withAttachment, String withoutAttachment, Set<String> releaseIds) {
+        List<Release> releasesSrc = new ArrayList<>();
+        if (withSourceAttachment != null && withSourceAttachment.equalsIgnoreCase("withSourceAttachment")) {
```

Comment:
After that use switch case instead of if else condition to filter the data

```
switch(filter)
    case withoutSourceAttachment:
             // write your code
             break;
    // other cases
```